### PR TITLE
added sample project yaml, added markdown parser, added projects/project page

### PR DESCRIPTION
**Pull request by awesominat**
this pr
1. adds a sample yaml showcasing the format
2. installs marked, a markdown parser
3. edited the Project schema to allow for teams, social info, markdown, description, images, publish date
4. added skeleton of projects page that loads all the projects from the yaml file
5. added project page that fetches exactly one project if project/[id] is accessed where id = slug (filename) of the yaml file

these are all necessary in order to create the projects and individual project page as per the figma design. all that's needed is to design the given and already rendered markdown & yaml of the projects.

tl;dr the PR handles fetching all the projects, a single project if /project/[id] is queried, and creates the first prototype of the project schema that it can be edited in the future

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,6 +13,8 @@
 /out/
 .vscode/
 
+package-lock.json
```
Comment (Cali0707) [inline .gitignore:16]: hey @awesominat normally I prefer to leave this in version control: https://stackoverflow.com/a/56254478/14216419

Comment (awesominat) [reply .gitignore]: done

**Review by Cali0707 — COMMENTED** [review]
Hey @awesominat this is a good start - I'm curious, why didn't you use something like react-markdown which would let us set custom components when we parse the markdown? An example of how to do this is here: https://blog.logrocket.com/mdx-integration-strategies-next-js/#react-markdown-mdx-integration

If there is a good reason for using marked, I'd love to hear it! Otherwise, could you switch to using react-markdown?

**Comment by awesominat** [comment]
no real reason, i can see how react-markdown is better. however, both marked and react-markdown allow us to have custom components that make it easier to style custom components.
marked does not allow you to set html without using dangerouslySetHtml unless the input is sanitized using some other module
marked is also standalone, while react-markdown has around 65 dependencies, some of which likely do the sanitizing for us
arguably, both work and allow us to do exactly what we want and how we want, with the sacrifice of a bit more setup vs downloading more dependencies

I've altered my branch to use react-markdown

**Comment by awesominat** [comment]
image

**Review by Cali0707 — APPROVED** [review]
Thanks for all your hard work @awesominat ! This is great!
